src/lib/compression/zstd_compressor.h

```diff
@@ -72,23 +72,27 @@ class zstd_compressor : public STREAM
 		return totalBytesRead;
 	}
 
-	unsigned pad(unsigned len)
+	size_t pad(unsigned len)
 	{
 		if (len < 9)
 			len = 9;
 
-		len -= 8;
-		std::array<char, 8> header = {0x50, 0x2A, 0x4D, 0x18, (char)(len&0xFF), (char)((len&0xFF00) >> 8), (char)((len&0xFF0000) >> 16), 0};
-		STREAM::write(header.data(), header.size());
+		// test what this actually does with ZSTD_isSkippableFrame and ZSTD_readSkippableFrame
+		// then, replace it with ZSTD_writeSkippableFrame...
+		std::string temp(len-8, '\0');
+		size_t writ = ZSTD_writeSkippableFrame(_compBuff.data(), _compBuff.size(), temp.data(), temp.size(), 0);
```

Comment:
This is the "unstable" static zstd api, which I think is why I was hesitant to use it initially. But it doesn't seem to have changed in 5+ years, so it's probably fine